openedx/core/djangoapps/oauth_dispatch/docs/decisions/0011-major-versioning.rst

```diff
@@ -0,0 +1,42 @@
+11. Handling Breaking Changes with Major Versions
+=================================================
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+The ``oauth_dispatch`` urls currently do not support versions. At this time, we would like to have the ability to iterate rapidly while working out details of how the API is called and the resulting payload of the JWT token.
+
+For background, please see:
+
+* `Use JWT as OAuth2 Tokens`_, where we decided to use JWTs for our oAuth2 tokens.
+
+.. _Use JWT as OAuth2 Tokens: 0003-use-jwt-as-oauth-tokens-remove-openid-connect.rst
+
+Decisions
+---------
+
+In order to support breaking changes, a major version will be introduced to the ``/access_token`` endpoint to allow the caller to request a particular major version.
+
+The major version will be supplied as a new query parameter. The parameter will look like *?v=N*, where N is the major version (e.g., '1', '2', etc.).
+
+* *Temporarily*, a default version will be allowed.
+
+  * At this time, a default of *v=1* will be assumed if the parameter is not provided.
+
+  * Once v1 is retired, and *v=1* is no longer an acceptable parameter, the parameter should become required and the endpoint should fail if no version is provided. This is consistent with other endpoints where the major version is required as part of the url path, making it more clear that major versions are supported.
```

Comment:
Just to check, "more clear that major versions are supported" or "more clear what major versions are supported"?

Reply:
The former.